tests/lib/tools/tests.backup

```diff
@@ -0,0 +1,59 @@
+#!/bin/sh -e
+
+show_help() {
+    echo "usage: tests.backup prepare"
+    echo "       tests.backup restore"
+}
+
+cmd_prepare() {
+    tar cf "${PWD}.tar" "$PWD"
+}
+
+cmd_restore() {
+    if [ -z "$PWD" ] || [ "$PWD" != "$(pwd)" ]; then
+        echo "tests.backup: cannot restore, wrong PWD varible: $PWD" >&2
+        exit 1
+    fi
+    if [ -f "${PWD}.tar" ]; then
+        # it is already checked that $PWD is not empty 
+        # shellcheck disable=SC2115
+        rm -rf "${PWD}"/*
```

Comment:
This is now doing things slightly different than before, why is that? I.e. in the existing code that was there before the dir was also removed which means any dot-files got removed. The new code does not have this property. Not saying this is right or wrong, just wondering why it is different from before?

Reply:
@mvo5 thanks for the review, you are right the new approach does not add any improvement. I'll restored the old approach.

Reply:
@mvo I had to revert that because if I delete the dir and call twice the prepare/restore in the same test it fails. I added a detail showing the error in the commit.

Reply:
Indeed, this is interessting. The inode of the working directory changes with the old approach so this is a nice fix. It will leave hidden files around, maybe something like:
```
find "$BACKUP_DIR" -name . -o -prune || xargs rm -rf
```

Reply:
updated